TFEngine.py

```diff
@@ -1344,48 +1345,81 @@ def _do_save(self):
         return False
     return True
 
-  def eval_model(self, output_file=None, output_per_seq_file=None):
+  def eval_model(self, output_file=None, output_per_seq_file=None, loss_name=None, output_per_seq_format=["seq_len", "score", "pos_score"]):
     """
     Eval the current model on the eval datasets (dev + eval, whatever is set).
     See also :func:`self.search` for performing beam search.
 
     :param str|None output_file: if given, will save the results to this file (total err/score for each dataset)
     :param str|None output_per_seq_file: if given, will save the err/score for each sequence
+    :param str|None loss_name: specifies the loss which will be written to output_file
+    :param list[str]|tuple[str]|None output_per_seq_format:
+      which properies of `loss_name` should be written to `output_per_seq_file`.
+      allowed_outputs = {"seq_tags", "seq_len", "score", "error", "pos_score", "pos_error"}
+
     :return: nothing
     """
+
     extra_fetches = None
-    results_per_seq = {}  # seq_tag -> dict[str,float]
+
     if output_per_seq_file:
-      extra_fetches = {}
+      allowed_outputs = {"seq_tags", "seq_len", "score", "error", "pos_score", "pos_error"}
+
+      assert isinstance(output_per_seq_format, (tuple, list))
```

Comment:
Maybe you also want to allow a single `str` here.
```suggestion
      assert isinstance(output_per_seq_format, (str, tuple, list))
      if isinstance(output_per_seq_format, str):
          output_per_seq_format = [output_per_seq_format]
```

Reply:
changed the default value in the config. List or tuple is required now